config/initializers/dfe_analytics.rb

```diff
@@ -0,0 +1,60 @@
+DfE::Analytics.configure do |config|
+  # Whether to log events instead of sending them to BigQuery.
+  #
+  # config.log_only = true
+
+  # Whether to use ActiveJob or dispatch events immediately.
+  #
+  # config.async = true
+
+  # Which ActiveJob queue to put events on
+  #
+  # config.queue = :default
+
+  # The name of the BigQuery table we’re writing to.
+  #
+
+  # The name of the BigQuery project we’re writing to.
+  #
+  # config.bigquery_project_id = ENV['BIGQUERY_PROJECT_ID']
+
+  # The name of the BigQuery dataset we're writing to.
+  #
+  # config.bigquery_dataset = ENV['BIGQUERY_DATASET']
+
+  config.bigquery_dataset = ENV["BIGQUERY_DFE_ANALYTICS_DATASET"] || Rails.application.credentials.BIGQUERY_DFE_ANALYTICS_DATASET
+
+  # Service account JSON key for the BigQuery API. See
+  # https://cloud.google.com/bigquery/docs/authentication/service-account-file
+  #
+  # config.bigquery_api_json_key = ENV['BIGQUERY_API_JSON_KEY']
+  config.bigquery_api_json_key = ENV["BIGQUERY_DFE_ANALYTICS_API_JSON_KEY"] || Rails.application.credentials.BIGQUERY_DFE_ANALYTICS_API_JSON_KEY
+
+  # Passed directly to the retries: option on the BigQuery client
+  #
+  # config.bigquery_retries = 3
+
+  # Passed directly to the timeout: option on the BigQuery client
+  #
+  # config.bigquery_timeout = 120
+
+  # A proc which returns true or false depending on whether you want to
+  # enable analytics. You might want to hook this up to a feature flag or
+  # environment variable.
+  config.enable_analytics = proc { Rails.env.staging? }
+
+
+  # The environment we’re running in. This value will be attached
+  # to all events we send to BigQuery.
+  #
+  config.environment = ENV.fetch('RAILS_ENV', 'development')
+
+  # A proc which will be called with the user object, and which should
+  # return the identifier for the user. This is useful for systems with
+  # users that don't use the id field.
+  #
+  # config.user_identifier = proc { |user| user&.id
+
+  # if part of any model name is migration, it wont be loaded
+ config.excluded_models_proc = proc { |x| x.to_s =~ /Migration::/ }
```

Comment:
Can the migration things not be added to the blocklist for now?

Reply:
The migration needs to be added, otherwise plugin will not work.
Currently we have models like `Statement` and `Migration::Statement`, which are pointing to 2 databases, but both are using `statements` table. The `dfe-analytics` operates on tables names so it gets confused by 2 tables that have the same name but are using 2 different schemas.

Excluding `Migration::` tables does not give much difference - all important data will be exported anyway.

Reply:
Ah, makes sense! Thanks.